addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -105,49 +127,29 @@ func main() {
 	// Use protobufs to improve performance.
 	config.ContentType = "application/vnd.kubernetes.protobuf"
 
-	k8s := nanny.NewKubernetesClient(*podNamespace, *deployment, *podName, *containerName, clientset, *useMetrics)
+	k8s := nanny.NewKubernetesClient(*podNamespace, *deployment, *podName, *containerName, clientset, *useMetrics, *runOnMaster)
 
 	nannyConfigurationFromFlags := &nannyconfigalpha.NannyConfiguration{
 		BaseCPU:       *baseCPU,
 		CPUPerNode:    *cpuPerResource,
 		BaseMemory:    *baseMemory,
 		MemoryPerNode: *memoryPerResource,
 	}
-	nannycfg, err := loadNannyConfiguration(*configDir, nannyConfigurationFromFlags)
-	if err != nil {
-		glog.Fatal(err)
-	}
-	glog.Infof("cpu: %s, extra_cpu: %s, memory: %s, extra_memory: %s", nannycfg.BaseCPU, nannycfg.CPUPerNode, nannycfg.BaseMemory, nannycfg.MemoryPerNode)
 
 	var resources []nanny.Resource
+	var nannyConfigUpdator nanny.NannyConfigUpdator
 
-	// Monitor only the resources specified.
-	if nannycfg.BaseCPU != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(nannycfg.BaseCPU),
-			ExtraPerResource: resource.MustParse(nannycfg.CPUPerNode),
-			Name:             "cpu",
-		})
-	}
-
-	if nannycfg.BaseMemory != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(nannycfg.BaseMemory),
-			ExtraPerResource: resource.MustParse(nannycfg.MemoryPerNode),
-			Name:             "memory",
-		})
-	}
-
-	if *baseStorage != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(*baseStorage),
-			ExtraPerResource: resource.MustParse(nannycfg.MemoryPerNode),
-			Name:             "storage",
-		})
+	if *runOnMaster {
```

Comment:
This updater is specific to KCP only use case. The old implementation is for addon resizer running in user nodes, and users will update the config and then manually restart metrics server pod. While the new updater fits only the case when addon resizer runs in masters, where customers cannot manually restart addon resizer so we need some watch mechanism. So I would prefer using runOnMaster flag to guard all these changes related to the scenario when running in masters. WDYT?

Reply:
In a few Kubernetes distros the control-plane VMs are accessible to users. The behavior is that the nanny config is watched instead of loaded only once, so I think I'd prefer a flag that describes that. But just a preference.

Reply:
I see. That makes sense. Let me use a separate flag then.